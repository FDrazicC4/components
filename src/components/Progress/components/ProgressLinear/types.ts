import { StyledComponent } from '@components/types';
import { Colors, Sizes } from '@components/Progress/types';


export type Variants = (
    | 'indeterminate'
    | 'determinate'
    | 'buffer'
    | 'query'
);


export interface Props extends StyledComponent {

    /**
     * The overall display of the progress indicator.
     */
    variant?: Variants,

    /**
     * The color of the progress indicator.
     */
    color?: Colors,

    /**
     * The size of the progress indicator.
     */
    size?: Sizes,

    /**
     * The value of the progress indicator for the determinate and static variants. Value between 0 and 100.
     */
    value?: number,

    /**
     * The value of the buffer indicator for the buffer variant. Value between 0 and 100.
     */
    bufferValue?: number,

}
