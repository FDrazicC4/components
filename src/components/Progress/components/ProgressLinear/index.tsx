import React from 'react';
import clsx from 'clsx';
import MuiProgressLinear from '@material-ui/core/LinearProgress';
import { Props } from './types';


/**
 * The `<ProgressLinear />` component renders a horizontal loading indicator.
 */
export const ProgressLinear = React.forwardRef(({
    variant = 'indeterminate',
    color = 'primary',
    size = 'inherit',
    bufferValue,
    className,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const classes = {
        bar: 'brew-ProgressLinear__bar',
        dashed: 'brew-ProgressLinear__dashedBar',
        bar2Buffer: 'brew-ProgressLinear__bufferBar',
    };


    const classNames = clsx(
        'brew-ProgressLinear',
        `brew-ProgressLinear--variant-${variant}`,
        `brew-ProgressLinear--color-${color}`,
        `brew-ProgressLinear--size-${size}`,
        className,
    );


    return (
        <MuiProgressLinear
            classes={classes}
            className={classNames}
            ref={ref}
            valueBuffer={bufferValue}
            variant={variant}
            {...otherProps} />
    );


});


ProgressLinear.displayName = 'ProgressLinear';


export default ProgressLinear;
