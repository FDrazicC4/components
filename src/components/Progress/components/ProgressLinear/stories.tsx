import React from 'react';
import { select, number } from '@storybook/addon-knobs';
import Flag from '@components/Flag';
import Progress from '@components/Progress';
import { Colors, Sizes } from '@components/Progress/types';
import { Variants } from './types';


export default {
    component: Progress.Circular,
    parameters: {
        componentSubtitle: <Flag color="success">Stable</Flag>,
    },
    title: 'Stable|Feedback/Progress/Linear',
};


export const Sandbox = (): React.ReactElement => {

    const color: Colors = select('color', [
        'primary',
        'secondary',
        'success',
        'error',
        'warning',
        'info',
        'light',
        'dark',
    ], 'primary');
    const size: Sizes = select('size', [
        'small',
        'medium',
        'large',
        'inherit',
    ], 'inherit');
    const variant: Variants = select('variant', ['indeterminate', 'determinate', 'buffer', 'query'], 'indeterminate');
    const value: number = number('value', 40);
    const bufferValue: number = number('bufferValue', 60);

    return (
        <Progress.Linear color={color} size={size} value={value} bufferValue={bufferValue} variant={variant} />
    );
};


export const General = (): React.ReactElement => <Progress.Linear />;
export const Indeterminate = (): React.ReactElement => <Progress.Linear variant="indeterminate" />;
export const Determinate = (): React.ReactElement => <Progress.Linear value={30} variant="determinate" />;
export const Buffer = (): React.ReactElement => <Progress.Linear value={40} valueBuffer={60} variant="buffer" />;
export const Query = (): React.ReactElement => <Progress.Linear variant="query" />;
