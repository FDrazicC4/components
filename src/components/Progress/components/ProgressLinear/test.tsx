import React from 'react';
import renderer from 'react-test-renderer';
import ProgressLinear from '.';


describe('ProgressLinear', () => {

    it('Renders correctly with defaults', () => {
        const tree = renderer
            .create(<ProgressLinear />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders correctly with property variants', () => {
        const tree = renderer
            .create((
                <React.Fragment>
                    <ProgressLinear color="primary" size="small" variant="indeterminate" />
                    <ProgressLinear color="secondary" size="medium" value={40} variant="determinate" />
                    <ProgressLinear bufferValue={60} color="tertiary" size="large" value={40} variant="buffer" />
                    <ProgressLinear color="success" variant="query" />
                    <ProgressLinear color="danger" />
                    <ProgressLinear color="warning" />
                    <ProgressLinear color="info" />
                </React.Fragment>
            ))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });


});
