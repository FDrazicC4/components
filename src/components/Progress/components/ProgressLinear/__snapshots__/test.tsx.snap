// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressLinear Renders correctly with defaults 1`] = `
<div
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-indeterminate brew-ProgressLinear--color-primary brew-ProgressLinear--size-inherit MuiLinearProgress-indeterminate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;

exports[`ProgressLinear Renders correctly with property variants 1`] = `
Array [
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-indeterminate brew-ProgressLinear--color-primary brew-ProgressLinear--size-small MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
      style={Object {}}
    />
  </div>,
  <div
    aria-valuenow={40}
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-determinate brew-ProgressLinear--color-secondary brew-ProgressLinear--size-medium MuiLinearProgress-determinate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
      style={
        Object {
          "transform": "translateX(-60%)",
        }
      }
    />
  </div>,
  <div
    aria-valuenow={40}
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-buffer brew-ProgressLinear--color-tertiary brew-ProgressLinear--size-large MuiLinearProgress-buffer"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-dashed brew-ProgressLinear__dashedBar MuiLinearProgress-dashedColorPrimary"
    />
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Buffer"
      style={
        Object {
          "transform": "translateX(-60%)",
        }
      }
    />
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-colorPrimary MuiLinearProgress-bar2Buffer brew-ProgressLinear__bufferBar"
      style={
        Object {
          "transform": "translateX(-40%)",
        }
      }
    />
  </div>,
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-query brew-ProgressLinear--color-success brew-ProgressLinear--size-inherit MuiLinearProgress-query"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
      style={Object {}}
    />
  </div>,
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-indeterminate brew-ProgressLinear--color-error brew-ProgressLinear--size-inherit MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
      style={Object {}}
    />
  </div>,
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-indeterminate brew-ProgressLinear--color-warning brew-ProgressLinear--size-inherit MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
      style={Object {}}
    />
  </div>,
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary brew-ProgressLinear brew-ProgressLinear--variant-indeterminate brew-ProgressLinear--color-info brew-ProgressLinear--size-inherit MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar brew-ProgressLinear__bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
      style={Object {}}
    />
  </div>,
]
`;
